frontend/src/main/web/lib/components/glossary/ActionCell.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import {PureRenderMixin} from 'react/addons';
+import { Button, Icon, Tooltip, OverlayTrigger } from 'zanata-ui';
+import Actions from '../../actions/GlossaryActions';
+import LoadingCell from './LoadingCell'
+import Comment from './Comment'
+import GlossaryStore from '../../stores/GlossaryStore';
+import StringUtils from '../../utils/StringUtils'
+import _ from 'lodash';
+
+var ActionCell = React.createClass({
+  propTypes: {
+    resId: React.PropTypes.string.isRequired,
+    info: React.PropTypes.string.isRequired,
+    rowIndex: React.PropTypes.number.isRequired,
+    canUpdateEntry: React.PropTypes.bool
+  },
+
+  mixins: [PureRenderMixin],
+
+  getInitialState: function() {
+    return this._getState();
+  },
+
+  _getState: function() {
+    return {
+      entry: GlossaryStore.getEntry(this.props.resId)
+    }
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    if (this.isMounted()) {
+      this.setState(this._getState());
+    }
+  },
+
+  _handleUpdate: function() {
+    Actions.updateGlossary(this.props.resId);
+  },
+
+  _handleCancel: function() {
+    Actions.resetEntry(this.props.resId);
+  },
+
+  _onUpdateComment: function (value) {
+    Actions.updateComment(this.props.resId, value);
+  },
+
+  render: function () {
+    var self = this;
+
+    if (self.props.resId === null || self.state.entry === null) {
+      return (<LoadingCell/>);
+    } else {
+      var isTransModified = self.state.entry.status.isTransModified;
+      var canUpdateComment = self.state.entry.status.canUpdateTransComment;
+      var isSaving = self.state.entry.status.isSaving === true;
+
+      var infoTooltip = <Tooltip id="info">{self.props.info}</Tooltip>;
+      var info = (
+        <OverlayTrigger placement='top' rootClose overlay={infoTooltip}>
+          <Icon className="cpri" name="info"/>
+        </OverlayTrigger>);
+
+      var updateButton = null, cancelButton = null,
+        comment = (
+          <Comment className="ml1/4" readOnly={!self.props.canUpdateEntry || !canUpdateComment || isSaving}
+            value={self.state.entry.transTerm.comment}
+            onUpdateCommentCallback={self._onUpdateComment}
+          />);
+
+      if(isSaving) {
+        return (<div>{info} {comment} <Button kind='primary' className="ml1/4" loading>Update</Button></div>);
+      }
+
+      if(isTransModified) {
+        updateButton = (<Button kind='primary' className='ml1/4' onClick={self._handleUpdate}>Update</Button>);
+        cancelButton = (<Button className='ml1/4' link onClick={self._handleCancel}>Cancel</Button>);
+      }
+
+      return (<div>{info} {comment} <div className='cdtargetib'>{updateButton} {cancelButton}</div></div>);
```

Comment:
Long lines of JSX here, please lay them out like HTML when they are not really simple. Also note that you only need a `(` before the JSX if it starts on the next line, otherwise you can just start it:

```
const thing = <span>Same line, no parens</span>
const bigThing = (
  <span>
    Starts on next line, so it needs the parens.
  </span>
)
```